circe-yaml/src/main/scala/io/circe/yaml/PrinterImpl.scala

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2016 circe
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.circe.yaml
+
+import io.circe.Json
+import io.circe.JsonNumber
+import io.circe.JsonObject
+import io.circe.yaml.PrinterBuilder.SnakeStringStyle
+import io.circe.yaml.common.Printer._
+import org.yaml.snakeyaml.DumperOptions
+import org.yaml.snakeyaml.emitter.Emitter
+import org.yaml.snakeyaml.nodes.MappingNode
+import org.yaml.snakeyaml.nodes.Node
+import org.yaml.snakeyaml.nodes.NodeTuple
+import org.yaml.snakeyaml.nodes.ScalarNode
+import org.yaml.snakeyaml.nodes.SequenceNode
+import org.yaml.snakeyaml.nodes.Tag
+import org.yaml.snakeyaml.resolver.Resolver
+import org.yaml.snakeyaml.serializer.Serializer
+
+import java.io.StringWriter
+import scala.collection.JavaConverters._
+
+class PrinterImpl(
```

Comment:
We should make this entire implementation private so we can extend it in a bin-compat way later.

Reply:
Changed. Applied also for v12 version